include/eve/arch/x86/tags.hpp

```diff
@@ -17,6 +17,15 @@
 
 namespace eve
 {
+  namespace detail
+  {
+# if defined(SPY_SIMD_IS_X86_AVX512)
+    static inline constexpr bool x86_logical_status = false;
+# else
+    static inline constexpr bool x86_logical_status = true;
+# endif
+  }
```

Comment:
What's this?

Reply:
This also seems a bit off.

`logical<wide<T, N, x86_256>>`

should not be bits. Or should it be bits anyways? 

Or should it also be ok to have

`logical<wide<char, eve::fixed<32>, _x86_512>`?

Reply:
When you activate AVX512, all logical values including the one from SSE/AVX have to be represented using the mmask registers. So I globally detect if AVX512 is up and fix this flag to true/false for later usage. It's a refactoring of not wanting the macro everywhere down